File: Reverse Word Order/reverse_word_order.cpp
/*
Exercise 15

Write a program (using functions!) that asks the user for a long string containing multiple words. 
Print back to the user the same string, except with the words in backwards order. For example, say I type the string:

-  My name is Michele

Then I would see the string:

- Michele is name My

shown back to me.
*/

#include <string>
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


vector <string> split (string string_to_split, string delimiter)
{
    vector <string> split_string;
    size_t pos = 0;
    string token;
    
    while ((pos = string_to_split.find(delimiter)) != string::npos) 
    {
        token = string_to_split.substr(0, pos);
        split_string.push_back(token);
        string_to_split.erase(0, pos + delimiter.length());
    }
    split_string.push_back(string_to_split);

    return split_string;
}

// string reverse_string(vector <string> vector_to_reverse)
// {
//     string rev_string;
//     int index = vector_to_reverse.size();
//     for (index; index != 0; index--)
//     {
//         cout << vector_to_reverse[index];
//         //rev_string.append(vector_to_reverse[index]);
//         // rev_string.append(" ");
//     }
//     return rev_string;
// }


int main()
{
    string test_string = {"My name is Michele"};
    string delimiter = {" "};

    vector <string> test_vector = split(test_string, delimiter);
    reverse(test_vector.rbegin(), test_vector.rend());

    for (string word : test_vector)
    {
        cout << word << " ";
    }

    
    return 0;
}
/*
TODO: This reverses the full string not the word order! 
*/

File: Reverse Word Order/reverse_word_order.py
"""
Exercise 15

Write a program (using functions!) that asks the user for a long string containing multiple words. 
Print back to the user the same string, except with the words in backwards order. For example, say I type the string:

-  My name is Michele

Then I would see the string:

- Michele is name My

shown back to me.
"""

# Modules

def reverse_string_1(string_to_reverse: str) -> str:
    """ Takes a string and reverses it. """

    reversed_wordorder = []
    for word in string_to_reverse.split():
        reversed_wordorder.append(word)
    return " ".join(reversed_wordorder[::-1])


def reverse_string_2(string_to_reverse: str) -> str:
    """ Takes a string and reverses it. """

    word_list = string_to_reverse.split()
    word_list.reverse()
    return " ".join(word_list)


def main():

    teststring = "My name is Michele"

    print(reverse_string_1(teststring))
    print(reverse_string_2(teststring))


if __name__ == "__main__":
    main()